templates/treat_ad.yaml

```diff
@@ -0,0 +1,61 @@
+Structural_Biology_Core_Folder: &struct_bio_folder
+  name: Structural_Biology_Core
+  type: Folder
+  next_level:
+    - Antibody_Validation_Data
+    - Assay_Protocols
+    - Assay_Results
+    - Cell_Lines
+    - Chemotypes
+    - Crystal_Structures
+    - Expression_Constructs
+    - Expression_Data
+    - Fragment_Screening_Results
+    - Lab_Notebooks
+    - Protein
+    - Protein_Expression_Methods
+    - TEP
+    
+Assay_Core_Folder: &assay_folder
+  name: Assay_Core
+  type: Folder
+  next_level:
+    - Assay_Results
+    - Compounds
+    - HTS_Protocols
+    - HTS_Results
+    - Isogenic_Cell_Line
+    - Lab_Notebooks
+    - Off_Target_Potency
+    - Secondary_Screen_Protocols
+    - Secondary_Screen_Result
+    
+MedChem_Core_Folder: &medchem_folder
+  name: MedChem_Core
+  type: Folder
+  next_level:
+    - Chemical_Probe_Data
+    - Chemical_Provenance
+    - Lab_Notebooks
+    - Structure_Activity_Relationship
+
+Gene_Folder: &gene_folder
+  - *struct_bio_folder
+  - *assay_folder
+  - *medchem_folder
+  - Bioinformatics Core
+
+name: TREAD_AD_YAML
+type: Project
+next_level:
+  - name: Genes
+    type: Folder
+    next_level:
+      - name: MSN
+        type: Folder
+        next_level: *gene_folder
```

Comment:
I'm assuming this references: `Gene_Folder: &gene_folder`.  I've never seen the `&` and `*` used this way.  I think it's clever, but I think we could potentially preserve the `*` as a wildcard.

Reply:
I think by itself, or preceded by a "." it can still be interpreted as a wildcard! It's worth looking into!

Reply:
phew! it hurts my brain to figure out that nesting.  Is there a way to generate the final output from this yaml?  Do yaml anchors even supported in json?  

I think a better alternative to this is to use a templating engine to do the nesting without resorting YAML anchors.

Reply:
There is!  The one in the PR description works for both visualizing the open document at the end of the YAML and the JSON representation of it!

Reply:
ahh neat, i didn't know that tool supported yaml anchors and i didn't know those can be converted to json.  It's still difficult to understand without the help of that tool but now i think it's bearable.

Reply:
I'm also new to YAML anchors and aliases. They're an elegant built-in solution to composing modular template chunks. 

One caveat (from what I can tell) is that the re-usable chunks have to be identical. For example, the gene name can't be used somewhere within the aliased structure (like an argument in a function). That said, I don't have a good sense yet of how common this use case would be. If it ends up being a common pattern, then we could explore using templating to "parametrize" aspects of the modular chunk. 

I'm guessing another caveat is that aliasing doesn't work with URLs, _e.g._ if the anchor is located in a central repository/catalog/registry. We could add special support for URLs (or contexts/namespaces à la JSON-LD).

Reply:
The use case is well-defined enough here that this makes sense.  I don't have a sense of other use cases either.  At the same time I don't see the name being a problem when it's user-defined - just name things accordingly?  

I didn't understand what you mean by `doesn't work with urls`.  But these repository could be a git repository.  I wrote a bigger description on the motivations that may make us want to do that rather than a database on slack when I posted the PR, but we can discuss it later today during the meeting too!